# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1790164849 -7200
# Parent  38162ee266a5a6a39cddb6b0dcc2a6bf4aa2d648
Bug 2073417 - Fix broken offline for POP AGAIN. Follow-up to unfortunate TB work in bug 2071585.

Based on the regression author's comment at:
https://phabricator.services.mozilla.com/D327465#11411081

diff --git a/mailnews/local/src/Pop3IncomingServer.sys.mjs b/mailnews/local/src/Pop3IncomingServer.sys.mjs
--- a/mailnews/local/src/Pop3IncomingServer.sys.mjs
+++ b/mailnews/local/src/Pop3IncomingServer.sys.mjs
@@ -128,8 +128,11 @@ export class Pop3IncomingServer extends 
         MailServices.pop3.CheckForNewMail(msgWindow, urlListener, inbox, this);
       }
     } catch (error) {
       this.performingBiff = false;
+      if (error.result == Cr.NS_MSG_ERROR_OFFLINE) {
+        return;
+      }
       throw error;
     }
   }
 
diff --git a/mailnews/local/src/Pop3Service.sys.mjs b/mailnews/local/src/Pop3Service.sys.mjs
--- a/mailnews/local/src/Pop3Service.sys.mjs
+++ b/mailnews/local/src/Pop3Service.sys.mjs
@@ -59,12 +59,10 @@ export class Pop3Service {
   }
 
   _getMail(downloadNewMail, msgWindow, urlListener, inbox, server) {
     if (Services.io.offline) {
-      throw Components.Exception(
-        "Cannot get mail while offline",
-        Cr.NS_MSG_ERROR_OFFLINE
-      );
+      Components.returnCode = Cr.NS_MSG_ERROR_OFFLINE;
+      return;
     }
     server.wrappedJSObject.withClient(client => {
       client.runningUri.msgWindow = msgWindow;
       client.urlListener = urlListener;
